include/dlaf/eigensolver/gen_to_std/mc/gen_to_std_L.h

```diff
@@ -0,0 +1,125 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <hpx/include/parallel_executors.hpp>
+#include <hpx/include/threads.hpp>
+
+#include "dlaf/blas_tile.h"
+#include "dlaf/common/index2d.h"
+#include "dlaf/common/pipeline.h"
+#include "dlaf/common/range2d.h"
+#include "dlaf/common/vector.h"
+#include "dlaf/communication/communicator_grid.h"
+#include "dlaf/communication/executor.h"
+#include "dlaf/communication/functions_sync.h"
+#include "dlaf/lapack_tile.h"
+#include "dlaf/matrix.h"
+#include "dlaf/matrix/distribution.h"
+#include "dlaf/util_matrix.h"
+
+namespace dlaf {
+namespace internal {
+namespace mc {
+
+// Implementation based on LAPACK Algorithm for the transformation from generalized to standard
+// eigenproblem (xHEGST)
+
+// Local implementation
+template <class T>
+void genToStd_L(Matrix<T, Device::CPU>& mat_a, Matrix<T, Device::CPU>& mat_l) {
+  constexpr auto Right = blas::Side::Right;
+  constexpr auto Left = blas::Side::Left;
+  constexpr auto Lower = blas::Uplo::Lower;
+  constexpr auto NonUnit = blas::Diag::NonUnit;
+  constexpr auto NoTrans = blas::Op::NoTrans;
+  constexpr auto ConjTrans = blas::Op::ConjTrans;
+
+  using hpx::util::unwrapping;
+
+  using hpx::threads::executors::pool_executor;
+  using hpx::threads::thread_priority_high;
+  using hpx::threads::thread_priority_default;
+
+  // Set up executor on the default queue with high priority.
+  pool_executor executor_hp("default", thread_priority_high);
+
+  // Set up executor on the default queue with default priority.
+  pool_executor executor_normal("default", thread_priority_default);
+
+  const SizeType m = mat_a.nrTiles().rows();
+  const SizeType n = mat_a.nrTiles().cols();
+
+  for (SizeType k = 0; k < n; ++k) {
+    const auto kk = LocalTileIndex{k, k};
+
+    // Direct transformation to standard eigenvalue problem of the diagonal tile
+    hpx::dataflow(executor_hp, unwrapping(tile::hegst<T, Device::CPU>), 1, Lower, mat_a(kk), mat_l(kk));
+
+    if (k != (n - 1)) {
+      const LocalTileIndex ai_start(k + 1, k);
+      const LocalTileIndex ai_end(m, k + 1);
+      const auto ai_panel = dlaf::common::iterate_range2d(ai_start, ai_end);
+
+      for (const auto& ik : ai_panel) {
+        hpx::dataflow(executor_normal, unwrapping(tile::trsm<T, Device::CPU>), Right, Lower, ConjTrans,
+                      NonUnit, 1.0, mat_l.read(kk), mat_a(ik));
+        hpx::dataflow(executor_normal, unwrapping(tile::hemm<T, Device::CPU>), Right, Lower, -0.5,
+                      mat_a.read(kk), mat_l.read(ik), 1.0, mat_a(ik));
+      }
+
+      const LocalTileIndex ti_start(k + 1, k + 1);
+      const LocalTileIndex ti_end(n, m);
+      const auto ti_trailing = dlaf::common::iterate_range2d(ti_start, ti_end);
+      for (const auto& ji : ti_trailing) {
```

Comment:
I don't understand why you iterate over `ji` and then use only the transposed `ij`.
Iterate directly on `ij`.